### ISSUE TITLE
How to obtain the guided backprop information from a filter in the intermediate layer of CNNs

### ISSUE THREAD

**Issue by jf003320018** [kundajelab/deeplift#5]
I try to find the way but the doc of the code is limited. So I do not know how to obtain the guided backprop information from a filter in the intermediate of CNNs.  I just want to know which function should I use and how to set the parameters of this function. Thank you.

**Comment by jf003320018** [comment]
Ok, it is my fault. I have not noticed your DeepLIFT + MoDISco tutorial.ipynb. 
It is seen that theano backend is used in DeepLIFT + MoDISco tutorial.ipynb. 
I want to confirm that can these results be reproduced by tensorflow backend?
Thank you.

**Comment by AvantiShri** [comment]
Yes, the API is the same on the tensorflow branch. To get the scores for an intermediate layer, you would replace find_scores_layer_idx with the index of the layer you want the scores for (in the TAL-GATA example it is 0 because you are finding scores for the input layer: https://github.com/kundajelab/deeplift/blob/tensorflow/examples/public/tal_gata/TAL-GATA%20simulation.ipynb).

By the way, if you are only interested in using Guided Backprop, then you may find it easier to simply override the gradients as illustrated in this code snippet: https://gist.github.com/falcondai/561d5eec7fed9ebf48751d124a77b087

You can then perhaps get the overridden gradients of your Keras model using something like this: https://github.com/kundajelab/deeplift/blob/tensorflow/tests/conversion/test_conv2d_model.py#L43-L45

**Comment by AvantiShri** [comment]
(I have not tried the alternate approach for guided backprop myself, though, so I can't promise it will work)

**Comment by jf003320018** [comment]
I will try it later. Thanks for your work.  ^^
